pkg/proxy/iptables/proxier.go

```diff
@@ -209,13 +209,44 @@ func newServiceInfo(svcPortName proxy.ServicePortName, port *api.ServicePort, se
 		loadBalancerStatus:       *helper.LoadBalancerStatusDeepCopy(&service.Status.LoadBalancer),
 		sessionAffinityType:      service.Spec.SessionAffinity,
 		stickyMaxAgeSeconds:      stickyMaxAgeSeconds,
-		externalIPs:              make([]string, len(service.Spec.ExternalIPs)),
-		loadBalancerSourceRanges: make([]string, len(service.Spec.LoadBalancerSourceRanges)),
+		externalIPs:              []string{},
+		loadBalancerSourceRanges: []string{},
 		onlyNodeLocalEndpoints:   onlyNodeLocalEndpoints,
 	}
 
-	copy(info.loadBalancerSourceRanges, service.Spec.LoadBalancerSourceRanges)
-	copy(info.externalIPs, service.Spec.ExternalIPs)
+	// Filter out the dual stack case.
+	for _, externalIP := range service.Spec.ExternalIPs {
+		if utilproxy.IsDualStackIP(externalIP, scm.isIPv6Mode) {
+			glog.Errorf("Dual stack not supported. Omitting IP %v in externalIPs for service %v/%v.", externalIP, service.Namespace, service.Name)
+			if scm.recorder != nil {
+				scm.recorder.Eventf(
+					&v1.ObjectReference{
+						Kind:      "Service",
+						Name:      service.Name,
+						Namespace: service.Namespace,
+						UID:       service.UID,
+					}, v1.EventTypeWarning, "KubeProxyDualStackUnsupported", "Unsupported dual stack externalIP: %v", externalIP)
```

Comment:
Before I push the updated commit, one more thing to point out --- emitting event like this means if there are 1,000 nodes, we will have 1,000 events, given that the hostname in EventSource is different. Would this be a problem? Or I can seek for a way to remove the hostname source so that they can be deduped.

Example events as below:
```
25s        25s         1         my-nginx                             Service               Warning   KubeProxyIncorrectIPVersion   kube-proxy, e2e-test-XXX-minion-group-p1ch   Incorrect IP version in loadBalancerSourceRanges: 2600:1900::/35
25s        25s         1         my-nginx                             Service               Warning   KubeProxyIncorrectIPVersion   kube-proxy, e2e-test-XXX-minion-group-5qks   Incorrect IP version in loadBalancerSourceRanges: 2600:1900::/35
25s        25s         1         my-nginx                             Service               Warning   KubeProxyIncorrectIPVersion   kube-proxy, e2e-test-XXX-minion-group-dz4x   Incorrect IP version in loadBalancerSourceRanges: 2600:1900::/35
```